### PR TITLE
Developer documentation and guide on adding new code

### LINKED ISSUE #62: Write guide on how to contribute for new users
Points to cover with an example (could be an example pull request)
- Write code (style recommendations)
- Write unit test (in tests/)

When adding/removing APIs
- Update `docs/source` files so that the changes shows in documentation.

### PR DESCRIPTION
**Pull request by AlexeyKurakin**
Resolves #62

### CODE REVIEW

File: docs/source/dev/adding_module.rst [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+* We allow to assign lambda, e.g. :code:`f = lambda x: x`
+
+Script :code:`./tests/run_linter.sh` automatically checks majority of code style violations.
+`PyCharm code formatter <https://www.jetbrains.com/help/pycharm/command-line-formatter.html>`_
+could be used to automatically reformat code.
+
+Writing unit tests
+------------------
+
+Unit tests are required for all code changes and new code of Objax library.
```
Comment (david-berthelot) [inline docs/source/dev/adding_module.rst:41]:
```suggestion
Unit tests are required for most code changes and new code of Objax library (but not for the `examples` folder).
```

File: docs/source/dev/adding_module.rst [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Other documentation is stored in `docs <https://github.com/google/objax/tree/master/docs>`_
+subdirectory of Objax repository.
+It uses `reST <https://docutils.sourceforge.io/rst.html>`_ as a markup language,
+and `Sphinx <https://www.sphinx-doc.org/>`_ automatically generates documentation
+for `<objax.readthedocs.io>`_.
+
+Docstrings
+^^^^^^^^^^
+
+All public facing classes, functions and class methods
```
Comment (david-berthelot) [inline docs/source/dev/adding_module.rst:66]:
```suggestion
All **public facing** classes, functions and class methods
```

File: docs/source/dev/adding_module.rst [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        Args:
+            logits: (batch, #class) tensor of logits.
+            labels: (batch, #class) tensor of label probabilities (e.g. labels.sum(axis=1) must be 1)
+
+        Returns:
+            (batch,) tensor of the cross-entropies for each entry.
+        """
+        return logsumexp(logits, axis=1) - (logits * labels).sum(1)
+
+If you only updating existing docstrings these changes will be automatically reflected
```
Comment (david-berthelot) [inline docs/source/dev/adding_module.rst:95]:
```suggestion
If you are only updating existing docstrings these changes will be automatically reflected
```

File: docs/source/dev/adding_module.rst [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+* `objax/random <https://github.com/google/objax/tree/master/objax/random>`_ contains routines
+  for random number generation.
+* `objax/zoo <https://github.com/google/objax/tree/master/objax/zoo>`_ is a "model zoo"
+  of various well-known neural network architectures.
+
+When writing code we follow PEP8 style guide with the following two exceptions:
+
+* We maximum line length to 120 characters.
+* We allow to assign lambda, e.g. :code:`f = lambda x: x`
+
```
Comment (david-berthelot) [inline docs/source/dev/adding_module.rst:33]:
```suggestion
* In addition we recommend trying to keep APIs ordered alphabetically when feasible within source code files.

Note: Remember to add your new APIs in `__all__` variable (also ordered alphabetically) at the top of the file to make them visible.
```

**Review by david-berthelot — CHANGES_REQUESTED** [review]
Minor typos/changes.

**Comment by AlexeyKurakin** [comment]
Updated, please take another look

**Review by david-berthelot — APPROVED** [review]

**Pull request merged**